<launch>
  
  <!-- joynode -->
  <node pkg="joy" 
        exec="joy_node"
        name="joy_node"
        output="screen">
    <!-- <param name="dev" value="/dev/input/js0"/> -->
    <param name="deadzone" value="0.05"/>
  </node>


  <arg name="input_ackermann_cmd" default="/ackermann_cmd"/>
  <arg name="output_control_cmd" default="/jetracer/cmd_drive"/>
  <arg name="output_trigger_topic" default="/rosbag2_recorder/trigger"/>
  <arg name="teleop_param"  default="$(find-pkg-share joy_manager)/config/teleop.param.yaml"/>
  <!-- joy manager node-->
  <node pkg="joy_manager"
        exec="joy_manager_node"
        name="joy_manager_node"
        output="screen">
        
    <param from="$(var teleop_param)"/>
    <!-- joy -->
    <remap from="/joy" to="/joy"/>
    <!-- 自律走行　コマンド -->
    <remap from="/ackermann_cmd" to="$(var input_ackermann_cmd)"/>
    <!-- jetracerへの入力 -->
    <remap from="/cmd_drive" to="$(var output_control_cmd)"/> 
    <!-- rosbag制御トリガ -->
    <remap from="/rosbag2_recorder/trigger" to="$(var output_trigger_topic)"/>
    <!-- offset トリガ -->
    <remap from="/steer_offset_inc" to="/steer_offset_inc"/>
    <remap from="/steer_offset_dec" to="/steer_offset_dec"/>
    <remap from="/speed_offset_inc" to="/speed_offset_inc"/>
    <remap from="/speed_offset_dec" to="/speed_offset_dec"/>
    
  </node>

</launch>
